core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +377,253 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
     }
   }
 
-  private List<String> toRoleNames(List<RoleEntity> roleEntities) {
-    return roleEntities.stream().map(RoleEntity::name).collect(Collectors.toList());
+  Role grantPrivilegesToRole(
+      String metalake, String role, MetadataObject object, List<Privilege> privileges) {
+    try {
+      AuthorizationPluginCallbackWrapper authorizationPluginCallbackWrapper =
+          new AuthorizationPluginCallbackWrapper();
+      Role updatedRole =
+          store.update(
+              AuthorizationUtils.ofRole(metalake, role),
+              RoleEntity.class,
+              Entity.EntityType.ROLE,
+              roleEntity -> {
+                List<SecurableObject> updateSecurableObjects =
+                    updateSecurableObjects(
+                        roleEntity.securableObjects(),
+                        object,
+                        oldObject -> {
+                          if (oldObject == null) {
+                            // Add a new securable object if there not exists the object in the role
+                            SecurableObject securableObject =
+                                SecurableObjects.parse(
+                                    object.fullName(),
+                                    object.type(),
+                                    Lists.newArrayList(privileges));
+
+                            authorizationPluginCallbackWrapper.setCallBack(
```

Comment:
Can we do this out of SQL transaction, this will hurt the performance and lead to unexpected behavior if we put it here.

Reply:
This place we only set the callback. After we execute SQL transaction, we execute the callbak.